File: java/src/main/java/io/hron/ast/DayRepeat.java
package io.hron.ast;

import java.util.List;

/**
 * A day-based repeat expression like "every day at 9:00" or "every 3 days at 09:00".
 *
 * @param interval the number of days between occurrences (1 for every day)
 * @param days the day filter (every, weekday, weekend, or specific days)
 * @param times the times of day to fire
 */
public record DayRepeat(int interval, DayFilter days, List<TimeOfDay> times)
    implements ScheduleExpr {
  /** Creates a new DayRepeat with defensive copy of times list. */
  public DayRepeat {
    times = List.copyOf(times);
  }
}

File: csharp/Hron/Ast/DayRepeat.cs
namespace Hron.Ast;

/// <summary>
/// A day-based repeat expression like "every day at 9:00" or "every 3 days at 09:00".
/// </summary>
/// <param name="Interval">The number of days between occurrences (1 for every day)</param>
/// <param name="Days">The day filter (every, weekday, weekend, or specific days)</param>
/// <param name="Times">The times of day to fire</param>
public sealed record DayRepeat(int Interval, DayFilter Days, IReadOnlyList<TimeOfDay> Times) : IScheduleExpr;
